arcgis-ios-sdk-samples/Scenes/Surface placements/SurfacePlacementsViewController.swift

```diff
@@ -17,65 +17,128 @@
 import UIKit
 import ArcGIS
 
-/// A view controller that manages the interface of the Surface Placements
-/// sample.
+/// A view controller that manages the interface of the Surface placements sample.
 class SurfacePlacementsViewController: UIViewController {
-    /// The scene displayed in the scene view.
-    let scene: AGSScene
+    // MARK: Instance properties
     
-    required init?(coder: NSCoder) {
-        scene = AGSScene(basemap: .topographic())
-        
-        // add base surface for elevation data
+    /// A label to show the value of the slider.
+    @IBOutlet weak var zValueLabel: UILabel!
+    /// The slider to change z-value of `AGSPoint` geometries, from 0 to 140 in meters.
+    @IBOutlet weak var zValueSlider: UISlider!
+    /// The switch to toggle the visibility of two draped mode graphics overlays.
+    @IBOutlet weak var drapedModeSwitch: UISwitch!
+    
+    /// The scene view managed by the view controller.
+    @IBOutlet var sceneView: AGSSceneView! {
+        didSet {
+            sceneView.scene = makeScene()
+            sceneView.setViewpointCamera(AGSCamera(latitude: 48.3889, longitude: -4.4595, altitude: 80, heading: 330, pitch: 97, roll: 0))
+            // Add graphics overlays of different surface placement modes to the scene.
+            graphicsOverlays.append(contentsOf: [
+                makeGraphicsOverlay(surfacePlacement: .drapedBillboarded),
+                makeGraphicsOverlay(surfacePlacement: .drapedFlat),
+                makeGraphicsOverlay(surfacePlacement: .relative),
+                makeGraphicsOverlay(surfacePlacement: .relativeToScene, offset: 2e-4),
+                makeGraphicsOverlay(surfacePlacement: .absolute)
+            ])
+            sceneView.graphicsOverlays.addObjects(from: graphicsOverlays)
+        }
+    }
+    
+    /// The graphics overlays of different surface placement modes.
+    var graphicsOverlays = [AGSGraphicsOverlay]()
+    /// A formatter to format z-value strings.
+    let zValueFormatter: NumberFormatter = {
+        let formatter = NumberFormatter()
+        formatter.numberStyle = .none
+        return formatter
+    }()
+    
+    // MARK: - Actions
+    
+    @IBAction func switchValueChanged(_ sender: UISwitch) {
+        // Toggle the visibility of two draped mode graphics overlays respectively.
+        graphicsOverlays.forEach { graphicOverlay in
+            switch graphicOverlay.sceneProperties!.surfacePlacement {
```

Comment:
Instead of looking for the draped graphic overlays each time, it would be cool to store them in a dictionary keyed by surface placement. Then updating their visibility would be much simpler:

```swift
overlaysBySurfacePlacement[.drapedFlat]?.isVisible = sender.isOn
overlaysBySurfacePlacement[.drapedBillboarded]?.isVisible = sender.isOn
```

Or better yet, just save off the draped overlays:

```swift
drapedOverlays.forEach { $0.isVisible = sender.isOn }
```

Reply:
It would be really great if SDK enums can conform to `CaseIterable`! 😄